### PR TITLE
Add readonly exhibit specific fields

### PR DESCRIPTION
**Pull request by tpendragon**

### CODE REVIEW

**Comment by tpendragon** [comment]
@cbeer @jkeck Any idea where these failures are coming from?

**Comment by cbeer** [comment]
Yeah, I think this is just the rubocop debacle. I was trying to wait it out, but maybe we should pin to `~> 0.37.2`?

File: db/migrate/20160329215014_add_readonly_to_custom_fields.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddReadonlyToCustomFields < ActiveRecord::Migration
+  def change
+    add_column :spotlight_custom_fields, :readonly_field, :boolean, default: false
```
Comment (cbeer) [inline db/migrate/20160329215014_add_readonly_to_custom_fields.rb]: Would `readonly` conflict with built-in AR methods?

File: blacklight-spotlight.gemspec [hunk 2]
```diff
@@ -57,7 +57,7 @@ these collections.)
   s.add_development_dependency 'rspec-activemodel-mocks'
   s.add_development_dependency 'rspec-collection_matchers'
   s.add_development_dependency 'capybara', '>= 2.5.0'
-  s.add_development_dependency 'rubocop', '~> 0.37'
+  s.add_development_dependency 'rubocop', '~> 0.37.2'
```
Comment (cbeer) [inline blacklight-spotlight.gemspec]: Would you mind changing this in a separate commit?

File: app/controllers/spotlight/catalog_controller.rb [hunk 3]
```diff
@@ -179,7 +179,7 @@ def uploaded_resource_params
     end
 
     def custom_field_params
-      current_exhibit.custom_fields.pluck(:field)
+      current_exhibit.custom_fields.where.not(readonly_field: true).pluck(:field)
```
Comment (cbeer) [inline app/controllers/spotlight/catalog_controller.rb]: Should we make this a scope? Either `current_exhibit.custom_fields.writeable.pluck` or `current_exhibit.writeable_custom_fields`?

**Comment by cbeer** [comment]
Nice, couple minor comments.

File: db/migrate/20160329215014_add_readonly_to_custom_fields.rb (hunk 1 above)
Comment (tpendragon) [inline db/migrate/20160329215014_add_readonly_to_custom_fields.rb]: Yeah

Comment (cbeer) [inline db/migrate/20160329215014_add_readonly_to_custom_fields.rb]: :ok_hand: figured as much. `readonly_field` is a little clunky, but I guess it'll have to do.

**Comment by tpendragon** [comment]
@cbeer Thanks for the comments, I think I've adjusted things now.

**Comment by cbeer** [comment]
:+1:

**Pull request merged**